spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDStreamParser.java

```diff
@@ -354,20 +385,27 @@ public DocumentParseResult parse(String[] document) {
 			DocumentParseResult result = new DocumentParseResult(document.length);
 			CrudRepository<BaseDefinition, String> transientRepository = new TransientDefinitionRepository();
 			StreamConfigParser parser = new StreamConfigParser(transientRepository);
+			int line = 1;
 			for (String nameAndDefinitionPair : document) {
 				try {
-
 					StreamNode stream = parser.parse(nameAndDefinitionPair);
 					String streamName = stream.getStreamName();
+					if (streamName == null) {
+						// Give the stream a placeholder name as none was supplied. Someone
+						// processing the parse result should know this can happen.
+						streamName = "UNKNOWN_" + line;
+					}
+					List<Exception> errorAccumulator = new ArrayList<Exception>();
 					List<ModuleDescriptor> moduleDescriptors = delegate.buildModuleDescriptors(streamName,
-							nameAndDefinitionPair, ParsingContext.stream, stream);
+							nameAndDefinitionPair, ParsingContext.partial_stream, stream, errorAccumulator);
```

Comment:
You have a keen eye for spotting that Ilaya :) I found the lack of positional info in those exceptions related to invalid options meant I couldn't really do a whole lot with them on the client at the moment. So I decided to go with partial_stream - it means the option validation isn't occurring but puts us in the same position as we are now (we aren't losing anything because we have never had option validation). Basically this PR was about doing the minimum to get us to something that works like it does today against our old endpoint in patched XD - we can take then take it forward from there after 1.2 is out and maybe enhance the exceptions with positions (the option validation one and the no such module one).